### PR TITLE
chore: remove underscore prefix from test packages

### PR DESCRIPTION
**Pull request by lkingland**
# Changes

- :broom: Remove underscore from test packages

The underscore causes various problems, such as in IDE autocomplete and code analys tasks, because this underscore  instructs go tooling to ignore the package.  The original purpose of this obfuscation was to work around a CI/CD problem which is no longer affecting us; so this PR removes the workaround.

/kind cleanup

**Release Note**

```release-note
Fixes an issue for developers where code in the test package would not be fully supported by some IDEs
```

### CODE REVIEW

**Comment by jrangelramos** [comment]
Thanks for going thru this @lkingland

**Comment by trisberg** [comment]
Probably due to this - https://spring.io/blog/2022/12/14/notice-of-permissions-changes-to-repo-spring-io-january-2023

**Review by zroubalik — APPROVED** [review]
/lgtm
/approve

File: templates/springboot/http/pom.xml [hunk 1]
```diff
@@ -18,9 +18,9 @@
 
   <properties>
     <java.version>11</java.version>
-    <spring-cloud.version>2021.0.5</spring-cloud.version>
-    <spring-native.version>0.12.2</spring-native.version>
-    <compiler-plugin.version>3.10.1</compiler-plugin.version>
+    <spring-cloud.version>2021.0.3</spring-cloud.version>
+    <spring-native.version>0.12.1</spring-native.version>
+    <compiler-plugin.version>3.8.1</compiler-plugin.version>
```
Comment (lance) [inline templates/springboot/http/pom.xml:23]: @lkingland the PR overall is great - thanks! Just curious why we are downgrading here.

**Pull request merged**